app/src/pages/trace/TracePage.tsx

```diff
@@ -12,20 +14,37 @@ import { TraceDetails } from "./TraceDetails";
  */
 export function TracePage() {
   const { traceId, projectId } = useParams();
+  const [searchParams] = useSearchParams();
   const navigate = useNavigate();
   const { rootPath, tab } = useProjectRootPath();
+  const selectedSpanNodeId = searchParams.get("selectedSpanNodeId");
+
+  // if we are focused on a particular span, use that as the subjectId
+  // otherwise, use the traceId
+  const paginationSubjectId = selectedSpanNodeId || traceId;
 
   return (
     <DialogContainer
       type="slideOver"
       isDismissable
-      onDismiss={() => navigate(`${rootPath}/${tab}`)}
+      onDismiss={() => {
+        navigate(`${rootPath}/${tab}`);
+      }}
     >
-      <Dialog size="fullscreen" title="Trace Details">
-        <TraceDetails
-          traceId={traceId as string}
-          projectId={projectId as string}
-        />
+      <Dialog
+        size="fullscreen"
+        title="Trace Details"
+        extra={<TraceDetailsPaginator currentId={paginationSubjectId} />}
+      >
+        <Suspense fallback={<Loading />}>
+          <TraceDetails
+            // blow out state when the paginationSubjectId changes
+            // some components are uncontrolled and will not update by themselves when the subjectId changes
+            key={paginationSubjectId}
```

Comment:
Can you help me understand this a bit more?

Reply:
Some parts of the trace details page do not reload their data properly when the spanId or traceId changes. Namely annotation inputs, because they are mostly uncontrolled and set a default value on mount.

I could try to be more surgical here, but it felt safer to just remount everything.

I can re-evaluate this since it's not necessarily the cleanest thing to manipulate the key prop like this